reference.py

```diff
@@ -61,9 +61,26 @@ def get_pubkey_from_input(vin: VinInfo) -> ECPubKey:
         if (pubkey.valid) & (pubkey.compressed):
             return pubkey
     if is_p2tr(vin.prevout):
+        witnessStack = vin.txinwitness.scriptWitness.stack
+        if (len(witnessStack) > 1):
```

Comment:
You should pop annex before determining if it's key path or script path spend.

From BIP-341:
* If there are at least two witness elements, and the first byte of the last element is 0x50, this last element is called annex a and is removed from the witness stack.
* If there is exactly one element left in the witness stack, key path spending is used.
* If there are at least two witness elements left, script path spending is used.